llpc/builder/llpcBuilderImplImage.cpp

```diff
@@ -984,6 +999,49 @@ Value* BuilderImplImage::CreateImageLoad(
                                   instName);
     }
 
+    // For 64-bit texel, only the first component is loaded, other components are filled in with (0, 0, 1). This
+    // operation could be viewed as supplement of the intrinsic call.
+    if (is64Bit && pOrigTexelTy->isVectorTy())
+    {
+        Value* pTexel = pResult;
+        if (isa<StructType>(pResultTy))
+        {
+            pTexel = CreateExtractValue(pResult, uint64_t(0));
+        }
+        pTexel = CreateBitCast(pTexel, getDoubleTy()); // Casted to double
+
+        Value* pZero = ConstantFP::get(getDoubleTy(), 0.0);
+        Value* pOne  = ConstantFP::get(getDoubleTy(), 1.0);
+        if (pOrigTexelTy->isIntOrIntVectorTy())
+        {
+            pZero = CreateBitCast(getInt64(0), getDoubleTy());
+            pOne  = CreateBitCast(getInt64(1), getDoubleTy());
+        }
```

Comment:
Put those into two sides of an if/else, to avoid creating the double FP constant when it's not needed.

Reply:
Actually, we only intend to support i64, I will remove the handling of double. Double ought to not be supported via R32G32.